Gems/ROS2/Code/Source/RobotImporter/Pages/PrefabMakerPage.cpp

```diff
@@ -60,5 +96,24 @@ namespace ROS2
     {
         return m_useArticulation->isChecked();
     }
+    AZStd::optional<AZ::Transform> PrefabMakerPage::getSelectedSpawnPoint() const
+    {
+        if (!m_spawnPointsInfos.empty())
+        {
+            int vectorIndex = m_spawnPointsComboBox->currentData().toInt();
+            AZStd::string mapKey = AZStd::string(m_spawnPointsComboBox->currentText().toStdString().c_str());
```

Comment:
This invokes both a constructor and a move operator, why not just initialize it directly?
```suggestion
            AZStd::string mapKey(m_spawnPointsComboBox->currentText().toStdString().c_str());
```